pkg/kubernetes/persistentvolumeclaim/v1alpha1/build.go

```diff
@@ -43,6 +43,21 @@ func (b *Builder) WithName(name string) *Builder {
 	return b
 }
 
+// WithGenerateName sets the GenerateName field of
+// PVC with provided value
+func (b *Builder) WithGenerateName(name string) *Builder {
+	if len(name) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build PVC object: missing PVC generateName"),
+		)
+		return b
+	}
+
+	b.pvc.object.GenerateName = name
```

Comment:
Can we have `b.pvc.object.GenerateName = name + "-"` as the generateName field does not append the `-` after the prefix on its own.

Reply:
The signature should change to accommodate delimiter.
Method should not append '-' on its own.
I would prefer not to change the signature and let the caller do the required stuff.
Will update the test logic to append suffix.
Thanks @shubham14bajpai for finding this

Reply:
Done